base/complex.jl

```diff
@@ -1028,3 +1028,68 @@ function complex(A::AbstractArray{T}) where T
     end
     convert(AbstractArray{typeof(complex(zero(T)))}, A)
 end
+
+function div(a::Complex{T}, b::Complex{V}, r::RoundingMode=RoundNearest) where {T<:Integer, V<:Integer}
+    R = promote_type(T, V)
+    a, b = Complex{R}(a), Complex{R}(b)
+    b̅ = conj(b)
+    # TODO: Handle overflow when calculating a*b̅
+    t = a*b̅
+    # TODO: Create checked_abs2(::Complex{<:Integer})
+    # TODO: Handle overflow when calculating the norm of complex numbers
+    abs2_b = abs2(b)
+    abs2_b < 0 && __throw_gcd_overflow(a, b)
```

Comment:
This also doesn't catch all cases of overflow, e.g.:
```julia-repl
julia> abs2(typemax(Int) + im*typemax(Int))
2
```
I think the best solution here is to add a `checked_abs2`, like you mentioned.